src/modules/edc-demo/components/contract-agreement-transfer-dialog/contract-agreement-transfer-dialog.component.ts

```diff
@@ -6,37 +6,48 @@ import {
   ContractAgreementService,
   DataAddressDto,
 } from '../../../edc-dmgmt-client';
+import {HttpDatasinkProperties} from '../../models/http-datasink-properties';
 import {AssetEntryBuilder} from '../../services/asset-entry-builder';
 import {NotificationService} from '../../services/notification.service';
+import {removeNullValues} from '../../utils/record-utils';
 import {ValidationMessages} from '../../validators/validation-messages';
 import {ContractAgreementTransferDialogData} from './contract-agreement-transfer-dialog-data';
 import {ContractAgreementTransferDialogForm} from './contract-agreement-transfer-dialog-form';
 import {ContractAgreementTransferDialogFormValue} from './contract-agreement-transfer-dialog-form-model';
 import {ContractAgreementTransferDialogResult} from './contract-agreement-transfer-dialog-result';
+import {ContractAgreementDatasinkFormBuilder} from './model/contract-agreement-datasink-form-builder';
+import {HttpDatasinkFormMapper} from './model/http-datasink-form-mapper';
 
 @Component({
   selector: 'edc-demo-contract-agreement-transfer-dialog',
   templateUrl: './contract-agreement-transfer-dialog.component.html',
-  providers: [ContractAgreementTransferDialogForm, AssetEntryBuilder],
+  providers: [
+    ContractAgreementTransferDialogForm,
+    AssetEntryBuilder,
+    ContractAgreementDatasinkFormBuilder,
+  ],
 })
 export class ContractAgreementTransferDialog implements OnDestroy {
   loading = false;
 
+  methods = ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'OPTIONS', 'HEAD'];
```

Comment:
GET method needs to be removed here. Http GET method cannot have a request body and data transfers will always have as request body.

Default should be POST